codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -213,6 +214,191 @@ private String generateHttpBindingSetter(Shape targetShape, Symbol targetSymbol,
+            GenerationContext context,
+            OperationShape operation
+    ) {
+        SymbolProvider symbolProvider = context.getSymbolProvider();
+        Model model = context.getModel();
+        GoWriter writer = context.getWriter();
+
+        Shape outputShape = model.expectShape(operation.getOutput()
+                .orElseThrow(() -> new CodegenException(
+                        "missing output shape for operation: " + operation.getId())));
+
+        HttpBindingIndex bindingIndex = context.getModel().getKnowledge(HttpBindingIndex.class);
+        Map<String, HttpBinding> bindingMap = bindingIndex.getResponseBindings(operation).entrySet().stream()
+                .filter(entry -> isRestBinding(entry.getValue().getLocation()))
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+
+        // do not generate if no HTTPBinding for operation output
+        if (bindingMap.size() == 0) {
+            return;
+        }
+
+        List<String> bindingMembers = new ArrayList<>(bindingMap.keySet());
+        bindingMembers.sort(String::compareTo);
+
+        Symbol outputSymbol = symbolProvider.toSymbol(outputShape);
+        writer.addUseImports(outputSymbol);
+
+        Symbol smithyHttpResponsePointableSymbol = SymbolUtils.createPointableSymbolBuilder(
+                "Response", GoDependency.SMITHY_HTTP_TRANSPORT).build();
+        writer.addUseImports(smithyHttpResponsePointableSymbol);
+
+        writer.addUseImports(SymbolUtils.createValueSymbolBuilder(null, GoDependency.FMT).build());
+
+        String functionName = ProtocolGenerator.getOperationDeserFunctionName(outputSymbol, getProtocolName());
+        writer.openBlock("func $L(v $P, response $P) error {", "}",
+                functionName, outputSymbol, smithyHttpResponsePointableSymbol,
+                () -> {
+                    writer.openBlock("if v == nil {", "}", () -> {
+                        writer.write("return fmt.Errorf(\"unsupported deserialization for nil %T\", v)");
+                    });
+
+                    writer.write("");
+
+                    for (int i = 0; i < bindingMembers.size(); i++) {
+                        HttpBinding binding = bindingMap.get(bindingMembers.get(i));
+                        writeRestDeserializerMember(writer, model, symbolProvider, binding);
+                        writer.write("");
+                    }
+
+                    writer.write("return nil");
+                });
+    }
+
+
+    private String generateHttpBindingsValue(Shape targetShape, Symbol targetSymbol, GoWriter writer, String operand) {
+        String value = "";
+
+        Symbol strConvPackageSymbol = SymbolUtils.createValueSymbolBuilder(null, GoDependency.STRCONV).build();
+        Symbol base64EncPackageSymbol = SymbolUtils.createValueSymbolBuilder(null, GoDependency.BASE64ENCODING).build();
+        Symbol privateProtoPackageSymbol = SymbolUtils.createValueSymbolBuilder(
```

Comment:
Again, you don't need to do this to import. Generally speaking, if you're creating a symbol with a null name then you probably don't want a symbol.